certs/certs.go

```diff
@@ -100,9 +112,86 @@ func ValidateRoot(certStore trustmanager.X509Store, root *data.Signed, gun strin
 		}
 	} else {
 		logrus.Debugf("found no currently valid root certificates for %s", gun)
+		logrus.Debugf("using trust_pinning config to bootstrap trust: %v", trustPinning)
```

Comment:
This method is too big. I think it might make sense to create an object that can validate a cert/GUN combo against a `TrustPinConfig`.

Something like:

```
trustPinChecker, err := notary.TrustPinChecker(trustPinConfig, GUN)
if err != nil {
 // blah
}
for each cert {
  trustPinChecker.checkCert(cert)
}
```

This object can encapsulate the logic about whether a given cert/GUN combo satisfies a PinningConfig. WDYT?